restli-tools/src/main/resources/apiVmTemplates/collection.vm

```diff
@@ -1,4 +1,138 @@
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*#
+
+package $spec.resource.namespace;
+
+import com.linkedin.data.DataMap;
+import com.linkedin.data.template.DataTemplateUtil;
+import com.linkedin.data.template.DynamicRecordMetadata;
+import com.linkedin.data.template.FieldDef;
+import com.linkedin.parseq.Engine;
+import com.linkedin.parseq.Task;
+import com.linkedin.parseq.function.Success;
+import com.linkedin.parseq.function.Try;
+import com.linkedin.restli.client.ParSeqRestliClient;
+import com.linkedin.restli.client.Response;
+import com.linkedin.restli.client.RestliRequestOptions;
+import com.linkedin.restli.common.ResourceMethod;
+import com.linkedin.restli.common.ResourceSpec;
+import com.linkedin.restli.common.ResourceSpecImpl;
+import com.linkedin.restli.common.RestConstants;
+import com.linkedin.restli.common.TypeSpec;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.EnumSet;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.CompletableFuture;
+import javax.annotation.Generated;
+import ${spec.entityClass.fullName};
+#foreach($method in $spec.restMethods)
+  #if($method.method.equals("get"))
+    import com.linkedin.restli.client.GetRequest;
+  #elseif($method.method.equals("batch_get"))
+    import com.linkedin.restli.client.BatchGetEntityRequest;
+    import com.linkedin.restli.client.response.BatchKVResponse;
+    import com.linkedin.restli.common.EntityResponse;
+    import com.linkedin.restli.internal.client.BatchEntityResponseDecoder;
+  #elseif($method.method.equals("create"))
+    #if($method.returnsEntity())
+      import com.linkedin.restli.client.CreateIdEntityRequest;
+      import com.linkedin.restli.common.IdEntityResponse;
+      import com.linkedin.restli.internal.client.IdEntityResponseDecoder;
+    #else
+      import com.linkedin.restli.client.CreateIdRequest;
+      import com.linkedin.restli.common.IdResponse;
+      import com.linkedin.restli.internal.client.IdResponseDecoder;
+    #end
+  #elseif($method.method.equals("batch_create"))
+    import com.linkedin.restli.common.CollectionRequest;
+    #if($method.returnsEntity())
+      import com.linkedin.restli.client.BatchCreateIdEntityRequest;
+      import com.linkedin.restli.common.CreateIdEntityStatus;
+      import com.linkedin.restli.common.BatchCreateIdEntityResponse;
+      import com.linkedin.restli.internal.client.BatchCreateIdEntityDecoder;
+    #else
+     import com.linkedin.restli.client.BatchCreateIdRequest;
+     import com.linkedin.restli.common.CreateIdStatus;
+     import com.linkedin.restli.common.BatchCreateIdResponse;
+     import com.linkedin.restli.internal.client.BatchCreateIdDecoder;
+    #end
+  #end
+#end
+@Generated("Generated from $spec.sourceIdlName")
+public class $spec.className {
```

Comment:
Another thing came to my mind is the name of the generated resource

In this example, Greetings, Greeting, Greeting(s)XXXX, those are actually quite confusing terms if we are going to mix another "generated" thing to such similar naming.

I personally suggest we clarify in some way that is is generated client. For example "GreetingsResource" or "GreetingsClient"

Reply:
The current naming is based on the RFC. The idea is to use "Greetings" for the fluent apis and then generate GreetingClient when they are combined into the universal client.

Reply:
I also didn't think that a big problem until I found in the code base and realize that how user can easily give simple and sometime random resouce name so things get confusing.( or maybe it is for this integration testing module?)

Also, at least for example
`find . -name "*greetings*java"`
Most of them are pretty self-explaining, non-conflicting whether auto-generated or not.

Reply:
This naming is only for our integ test module. In real world, service names would be meaningful.